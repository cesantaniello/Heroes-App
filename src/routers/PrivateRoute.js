import { useContext } from "react";
import { Navigate, useLocation } from "react-router-dom";
import { AuthContext } from "../auth/authContext";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    Navigate: () => <span>Exit</span>,
}));

export const PrivateRoute = ({children}) => {
    
    const {user} = useContext(AuthContext);
    const {pathname, search} = useLocation();
    
    localStorage.setItem('lastPath', pathname + search);

    return user.logged
        ? children
        : <Navigate to="/login"/>
}
