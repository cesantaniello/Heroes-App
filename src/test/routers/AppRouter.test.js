import { mount } from "enzyme";
import { AuthContext } from "../../auth/authContext";
import { AppRouter } from "../../routers/AppRouter";

describe('Tests on <AppRouter />', () => {

    const contextValue = {
        user: {logged: false}
    }

    test('Should show a login if the user is not authenticated', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Login');
    })
})