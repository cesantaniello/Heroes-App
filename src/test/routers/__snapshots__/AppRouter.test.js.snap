// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> Should show a login if the user is not authenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
